### PR TITLE
added step definition locations

### PR DESCRIPTION
**Pull request by AlexGalichenko**
added log displaying

### CODE REVIEW

File: src/formatter.js [hunk 1]
```diff
@@ -71,6 +77,11 @@ class PrettyFormatter extends Formatter {
         }
     }
 
+    readStepDefinition(stepDefinition) {
+        const definition = stepDefinition.stepDefinition ?? stepDefinition.hook;
+        this.stepDefinitions[definition.id] = definition.sourceReference.uri + ':' + definition.sourceReference.location.line;
```
Comment (nomid0612) [inline src/formatter.js:82]: it is better replace to `${definition.sourceReference.uri}:${definition.sourceReference.location.line}`

File: src/formatter.js [hunk 2]
```diff
@@ -126,6 +142,12 @@ class PrettyFormatter extends Formatter {
 
     drawStep(step) {
         let line = this.indent + chalk.bold(this.keywords[step.type] ?? '*') + ' ' + step.stepText;
+        line += ' ' + chalk.gray(step.location);
+        if (this.showLogs) {
+            for (const log of step.logs) {
+                line += '\n' + this.indent + log.body;
```
Comment (nomid0612) [inline src/formatter.js:148]: it is similar to comment above with `${}`

File: src/formatter.js [hunk 3]
```diff
@@ -126,6 +142,12 @@ class PrettyFormatter extends Formatter {
...
+            }
+        }
         if (step.argument && step.argument.dataTable) {
             line += '\n' + this.drawDataTable(step.argument.dataTable)
```
Comment (nomid0612) [inline src/formatter.js:152]: it is similar to comment above with ${}

File: src/formatter.js (hunk 1 above)
Comment (AlexGalichenko) [reply src/formatter.js:82]: I'll refactor it later

File: src/formatter.js (hunk 2 above)
Comment (AlexGalichenko) [reply src/formatter.js:148]: all concats via + will be refactored later

File: src/formatter.js (hunk 3 above)
Comment (AlexGalichenko) [reply src/formatter.js:152]: all concats via + will be refactored later

**Review by nomid0612 — APPROVED** [review]

**Pull request merged**